docs(DateTimePicker): move to Storybook (#408)
zextras/carbonio-design-system: src/components/inputs/datetimePicker/DateTimePicker.mdx

import { Canvas, Controls, Description, Meta, Primary, Stories, Subheading, Title } from '@storybook/blocks';
import * as DateTimePickerStories from './DateTimePicker.stories';
import WithCustomInput from './DateTimePicker.stories.customInput?raw'
import WithDefaultLocale from './DateTimePicker.stories.defaultLocale?raw'
import WithRegisteredLocale from './DateTimePicker.stories.registeredLocale?raw'
import WithLocaleObject from './DateTimePicker.stories.localeObject?raw'

<Meta of={DateTimePickerStories} />

<Title />

DateTimePicker is an extension of the picker provided by react-datepicker library. The extension is made to style the
library component in a ds-style and to give a default input ready to be used.

The props shown here are the one defined for the extension, but it is possible to use all the props of the library itself,
where not specified here (in this case, they are being overwritten by our component, with a possible different meaning).
Refer to the official documentation to get a list of them [https://github.com/Hacker0x01/react-datepicker]()

> By default, the DateTimePicker provide 2 types of input, a text input and a chip input.

Reviewer: ```suggestion
By default, the DateTimePicker provides 2 types of input, a text input and a chip input.
```

The behaviour of the two components can differ a bit because of how react-datepicker works under the hood.

Examples of the library component are visible here [https://reactdatepicker.com/]()

<Canvas of={DateTimePickerStories.Default} story={{ height: '350px' }} withToolbar />

<Controls />

### With Custom input

> When defining a custom input, it is important to create a component which accept a ref. React-datepicker offer the possibility to set

Reviewer: ```suggestion
When defining a custom input, it is important to create a component that accepts a ref. React-datepicker offers the possibility to set
```

a different name for the prop of the input component in charge of accepting the ref object. This prop is `customInputRef`.

> On most of the cases, creating a component with React.forwardRef is sufficient.

Reviewer: ```suggestion
In most of the cases, creating a component with React.forwardRef is enough.
```
